### ISSUE TITLE
Error reproducing ego_progress results on Navhard

### ISSUE THREAD

**Issue by Kevin6liu6** [valeoai/DrivoR#47]
Hi there,

Thank you all for your dedication and great work. I was trying to run evaluation on your released weights for nav2. I followed all instructions  in https://github.com/valeoai/DrivoR/issues/13. At first, I got 0s for several submetrics and a very weird 0.0008xxx overall EPDMS. Then I fixed this issue by looking up the issue section of navsim2 repo (it was caused by numpy version mismatch). Now, I can get everything else right except for ego_progress:

<img width="1433" height="63" alt="Image" src="https://github.com/user-attachments/assets/cf998196-9dd6-4213-98a4-8df77078c75e" />

<img width="835" height="188" alt="Image" src="https://github.com/user-attachments/assets/a7aced56-86c4-4a8c-8f1d-60c86731f0e7" />

Any idea what might be wrong?

**Comment by tongnie** [comment]
same question.

**Comment by yihongXU** [comment]
Hi, it seems that you have a much higher EP? Could you tell me what did you do to eval Nav2? Is it related to https://github.com/autonomousvision/navsim/issues/151?

**Comment by yihongXU** [comment]
And maybe try to re-cache your metrics using navsim repo? but EP is related to PDM-lite trajectories.

**Comment by yihongXU** [comment]
Could it be related to H100 GPUs? Both when you do your caching and eval.

**Comment by Kevin6liu6** [comment]
> Hi, it seems that you have a much higher EP? Could you tell me what did you do to eval Nav2? Is it related to [autonomousvision/navsim#151](https://github.com/autonomousvision/navsim/issues/151)?

Nope, it's nothing to do with https://github.com/autonomousvision/navsim/issues/151. I cloned the repo of GTRS, and follow the instructions of https://github.com/valeoai/DrivoR/issues/13, substituted the agent, config and 3 attached .py scipts. At first, I got 0s for many subscores, and then I installed numpy==1.26.4 and everything looks ok expect for EP.

**Comment by Kevin6liu6** [comment]
> And maybe try to re-cache your metrics using navsim repo? but EP is related to PDM-lite trajectories.

Sure, I will try to re-cache it

**Comment by Kevin6liu6** [comment]
> Could it be related to H100 GPUs? Both when you do your caching and eval.

I used H20 for caching and eval, is that ok?

**Comment by yihongXU** [comment]
Or you could try the official Nav2 repo and use the GPU version of run_pdm_score_gpu_v2.py:
[run_pdm_score_gpu_v2.py](https://github.com/user-attachments/files/24762954/run_pdm_score_gpu_v2.py)
[agent_lightning_module.py](https://github.com/user-attachments/files/24762955/agent_lightning_module.py)

**Comment by yihongXU** [comment]
> > And maybe try to re-cache your metrics using navsim repo? but EP is related to PDM-lite trajectories.
> 
> Sure, I will try to re-cache it

Yes, please try to re-cache with the official Nav2 repo for navhard.

**Comment by yihongXU** [comment]
> > Could it be related to H100 GPUs? Both when you do your caching and eval.
> 
> I used H20 for caching and eval, is that ok?

We tried on H100, it looks fine. btw, make sure you're using the correct ckpt: https://github.com/valeoai/DrivoR/releases/download/Scaling/nav2_30_epochs_with_134k_simscale_85ktrain_54.6.pth

**Comment by yihongXU** [comment]
You can also try to submit your result  to the navhard server, we obtained the correct result as we evaluated: [run_create_submission_pickle_warmup_gpu.py](https://github.com/user-attachments/files/29335332/run_create_submission_pickle_warmup_gpu.py)

[default_run_create_submission_pickle_gpu.yaml](https://github.com/user-attachments/files/29339867/default_run_create_submission_pickle_gpu.yaml)

> TEAM_NAME=
AUTHORS=
EMAIL=
INSTITUTION=
COUNTRY=
TRAIN_TEST_SPLIT=navhard_two_stage
CACHE_PATH=$NAVSIM_EXP_ROOT/navhard_two_stage_metric_cache
SYNTHETIC_SENSOR_PATH=$OPENSCENE_DATA_ROOT/navhard_two_stage/sensor_blobs
SYNTHETIC_SCENES_PATH=$OPENSCENE_DATA_ROOT/navhard_two_stage/synthetic_scene_pickles
export SUBSCORE_PATH=$NAVSIM_EXP_ROOT
CHECKPOINT=[nav2_30_epochs_with_134k_simscale_85ktrain_54.6.pth](https://github.com/valeoai/DrivoR/releases/download/Scaling/nav2_30_epochs_with_134k_simscale_85ktrain_54.6.pth)
EXPERIMENT=drivoR_nav2
AGENT=drivoR
python $NAVSIM_DEVKIT_ROOT/navsim/planning/script/run_create_submission_pickle_warmup_gpu.py \
    train_test_split=$TRAIN_TEST_SPLIT \
    experiment_name=$EXPERIMENT \
    metric_cache_path=$CACHE_PATH \
    synthetic_sensor_path=$SYNTHETIC_SENSOR_PATH \
    synthetic_scenes_path=$SYNTHETIC_SCENES_PATH \
    agent=$AGENT \
    agent.checkpoint_path=$CHECKPOINT \
    agent.config.refiner_ls_values=0.0 \
    agent.config.image_backbone.focus_front_cam=false \
    agent.config.one_token_per_traj=true \
    agent.config.proposal_num=64 \
    agent.config.refiner_num_heads=1 \
    agent.config.tf_d_model=256 \
    agent.config.tf_d_ffn=1024 \
    agent.config.area_pred=false \
    agent.config.agent_pred=false \
    agent.config.ref_num=4 \
    agent.config.noc=10 \
    agent.config.dac=13 \
    agent.config.ddc=6 \
    agent.config.ttc=14 \
    agent.config.ep=15 \
    agent.config.comfort=2 \
    agent.loss.prev_weight=0.0 \
    team_name=$TEAM_NAME \
    authors=$AUTHORS \
    email=$EMAIL \
    institution=$INSTITUTION \
    country=$COUNTRY \
    +seed=2

**Comment by Kevin6liu6** [comment]
> > > And maybe try to re-cache your metrics using navsim repo? but EP is related to PDM-lite trajectories.
> > 
> > 
> > Sure, I will try to re-cache it
> 
> Yes, please try to re-cache with the official Nav2 repo for navhard.

I recached navhard and now all the scores matches with the paper's results. Thank you!

> > > And maybe try to re-cache your metrics using navsim repo? but EP is related to PDM-lite trajectories.
> > 
> > 
> > Sure, I will try to re-cache it
> 
> Yes, please try to re-cache with the official Nav2 repo for navhard.

I recached navhard set and now everything matches with the results in the paper. Thank you! 
To whom it may concern: if you have the numpy version problem, remember to recache after upgrading your numpy.
